### PR TITLE
Add configurable series selection to upload step

### LINKED ISSUE #777: Add dropdown selection of series or show series name if series ID has been given
If the series ID has been passed as GET parameter (for example), then the upload dialogue could show the series name. 

If it hasn't been given, we can let the user decide in which series (if any) to upload the recording. This selection should only contain series that the user has write access to (those can now be queried thanks to https://github.com/opencast/opencast/pull/1548).

### PR DESCRIPTION
**Pull request by LukasKalbertodt**
Closes #777 
Replaces #992 

This is #992 rebased on the redesign. Well, since so much changed, it's almost more of a reimplementation than just a rebase. 

If `upload.seriesId` is set, then that series is pre-selected in the series-selector, but the user can still change it. If organizations want the old behavior, they need to set `upload.seriesField=hidden`. 

| [Test Deployment](https://test.studio.opencast.org/2023-10-18_10-06-50-pr1143-6559335813/) |
| --- |

@geichelberger Can you review and/or test this? If it meets all your requirements? Or is someone else better suited to answer the latter question? I can also find someone here at ELAN to do the code review, if you don't have the time. Just let me know!

### CODE REVIEW

**Comment by majosch** [comment]
@LukasKalbertodt Thank you. At University of Vienna we are desperately in need for this feature :sweat_smile: 

We did some testing on our "upstream" Opencast installation, that is built from develop-branch - and on our "staging" instance with Opencast 14.x and backported [PR4651](https://github.com/opencast/opencast/pull/4651) . Works as expected on both instances - also if deployed as "standalone".

File: CONFIGURATION.md [hunk 1]
```diff
@@ -102,6 +102,11 @@ further below for information on that.
 # Default: 'required'.
 #presenterField = 'required'
 
+# Specifies if the series field should be 'hidden', 'optional' or 'required'.
+# Default: 'required'.
+#seriesField = 'required'
```
Comment (geichelberger) [inline CONFIGURATION.md:107]: I guess `required` is the most common use case.

**Review by geichelberger — APPROVED** [review]
LGTM

File: CONFIGURATION.md (hunk 1 above)
Comment (LukasKalbertodt) [reply CONFIGURATION.md:107]: Oh wow good catch! So the code defaults to `optional`, so these docs are incorrect. Probably copy&paste error on my side... for some reason. I would actually argue to make it `optional` by default, as currently implemented in code. Will push that change.

**Review by geichelberger — APPROVED** [review]

**Comment by LukasKalbertodt** [comment]
I created a release and opened the PR against Opencast to update Studio, such that this feature will be included in the upcoming OC 15: https://github.com/opencast/opencast/pull/5418

**Pull request merged**